src/herder/TxSetFrame.cpp

```diff
@@ -1259,13 +1241,13 @@ TxSetPhaseFrame::Iterator::operator++()
         throw std::runtime_error("TxPhase iterator out of bounds");
     }
     ++mTxIndex;
-    if (mTxIndex >= mStages[mStageIndex][mThreadIndex].size())
+    if (mTxIndex >= mStages[mStageIndex][mClusterIndex].size())
```

Comment:
Should the if condition above also check `mClusterIndex >= mStages[mStageIndex].size()`? It'll be checked in `operator*()`, but can't hurt to add it here to be thorough.

Reply:
Done.